src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -0,0 +1,274 @@
+                        return Promise.resolve('Stale request');
+                    }
+
+                    values.forEach(function (datum) {
+                        this.updateHistory(datum);
+                    }, this);
+
+                    this.updateValues(values[values.length - 1]);
+                }.bind(this));
+        },
+        scrollToBottom() {
+            if (this.isPaused || !this.$refs.thumbsWrapper || !this.autoScroll) {
+                return;
+            }
+
+            const scrollHeight = this.$refs.thumbsWrapper.scrollHeight || 0;
+            if (!scrollHeight) {
+                return;
+            }
+
+            setTimeout(() => this.$refs.thumbsWrapper.scrollTop = scrollHeight, 0);
+        },
+        setSelectedImage(image) {
+            this.imageUrl = this.getImageUrl(image);
+            this.time = this.getTime(image);
+            this.paused(true);
+            this.unselectAllImages();
+            image.selected = true;
+        },
+        getElSize() {
+            if (!this.$el) {
+                return;
+            }
+
+            this.elementHeight = this.$el.parentElement.offsetHeight;
+            return this.elementHeight;
+        },
+        stopListening() {
+            if (this.unsubscribe) {
+                this.unsubscribe();
+                delete this.unsubscribe;
+            }
+        },
+        subscribe(domainObject) {
+            this.date = ''
+            this.imageUrl = '';
+            this.openmct.objects.get(this.keystring)
+                .then(function (object) {
+                    const metadata = this.openmct
+                        .telemetry
+                        .getMetadata(this.domainObject);
+                    this.timeKey = this.openmct.time.timeSystem().key;
+                    this.timeFormat = this.openmct
+                        .telemetry
+                        .getValueFormatter(metadata.value(this.timeKey));
+                    this.imageFormat = this.openmct
+                        .telemetry
+                        .getValueFormatter(metadata.valuesForHints(['image'])[0]);
+                    this.unsubscribe = this.openmct.telemetry
+                        .subscribe(this.domainObject, function (datum) {
+                            this.updateHistory(datum);
+                            this.updateValues(datum);
+                        }.bind(this));
+
+                    this.requestHistory(this.openmct.time.bounds());
+                }.bind(this));
+        },
+        unselectAllImages() {
+            this.imageHistory.forEach(image => image.selected = false);
+        },
+        updateHistory(datum) {
+            if (this.datumMatchesMostRecent(datum)) {
+                return false;
+            }
+
+            const index = _.sortedIndex(this.imageHistory, datum, this.timeFormat.format.bind(this.timeFormat));
+            this.imageHistory.splice(index, 0, datum);
+
+            return true;
+        },
```

Comment:
Why is this function returning a boolean? Can you just set `imageHistory` if `this.datumMatchesMostRecent(datum)` is not true?

Reply:
updated.